### PR TITLE
refactor: datasets client to extend browsable client

### PR DESCRIPTION
**Pull request by jywadhwani**
Also deprecating an obsolete file `DatasetUtils.java`

## Checklist
- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: gms/client/src/main/java/com/linkedin/common/client/DatasetsClient.java [hunk 1]
```diff
@@ -18,9 +18,4 @@ protected DatasetKey toDatasetKey(@Nonnull DatasetUrn urn) {
                 .setOrigin(urn.getOriginEntity())
                 .setPlatform(urn.getPlatformEntity());
     }
-
-    @Nonnull
-    protected DatasetUrn toDatasetUrn(@Nonnull DatasetKey key) {
-        return new DatasetUrn(key.getPlatform(), key.getName(), key.getOrigin());
-    }
 }
```
Comment (keremsahin1) [inline gms/client/src/main/java/com/linkedin/common/client/DatasetsClient.java:21]: Probably rename this as BaseDatasetsClient?

**Review by keremsahin1 — COMMENTED** [review]
Please also make sure to test this E2E

File: gms/client/src/main/java/com/linkedin/common/client/DatasetsClient.java (hunk 1 above)
Comment (jywadhwani) [reply gms/client/src/main/java/com/linkedin/common/client/DatasetsClient.java:21]: I will prefer to move the method `toDatasetKey` in `DatasetsClient` to a utility file, will leave it for another PR.

**Comment by jywadhwani** [comment]
@keremsahin1  tested this E2E - search, browse and autocomplete work as expected for datasets.

**Review by keremsahin1 — APPROVED** [review]

**Pull request merged**